photon-client/src/main/scala/com/linkedin/photon/ml/io/deprecated/GLMSuite.scala

```diff
@@ -344,13 +344,22 @@ class GLMSuite(
           1
         }
 
+      assert(weight >= 0)
```

Comment:
Do we want to fail on negative weights, or do we want to filter them? If the former, this should be a `require` instead of an assert. If the latter, do we want to do so silently or log info about it?